teamup-server/src/main/java/com/project/teamup/service/validator/UserValidator.java

```diff
@@ -0,0 +1,44 @@
+package com.project.teamup.service.validator;
+
+import com.project.teamup.model.User;
+import com.project.teamup.model.UserSkill;
+import com.project.teamup.model.UserSkillLevel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Please fill out this class with further functions to validate other attributes.
+ * @author Sebastian
+ */
+@Component
+public class UserValidator {
+    @Autowired
+    private TechnologyValidator technologyValidator;
+
+    // Add your validation functions here.
+    public boolean validateObject(User user){
+        return validateUserSkills(user.getSkills());
+    }
+
+    /**
+     * Checks the validity of the {@link UserSkill} object.
+     * Method is set to 'public' for testing purposes.
+     * @param userSkillList is the list of Skills tied to the user.
+     * @return true if valid, false if not
+     */
+    public boolean validateUserSkills(List<UserSkill> userSkillList){
+        if (userSkillList != null && userSkillList.size() != 0){
```

Comment:
userSkillsList can be empty or null,  skills addition is not mandatory from what I understood. 
For the user skills validation it would be better to create a method that validates if a skill has a technology and that tehnology has a technology area. And use tha method when itterating on the list.
So for now as I mentioned above I think that annotation validations on each model field would be better for our case(to have an readable code and easy to mantain).

Reply:
I am aware of the fact that we allow the user to not set any skills, that's why I only validate the objects inside the list, if the list is not empty or not null. If it is empty or null, I just return true, mocking the fact that it's valid.

Reply:
I'll see what I can do to include Annotations and not business logic.